.github/workflows/build_magma_dep.yml

```diff
@@ -0,0 +1,73 @@
+---
+name: "Build Nettle and Upload to Artifactory"
+on:
+  workflow_dispatch:
+    inputs:
+      repo:
+        description: 'Artifactory Repository'
+        required: false
+        default: 'debian-test'
+      distribution:
+        description: 'Distribution'
+        required: false
+        default: 'focal-ci'
+      dry_run:
+        description: 'Dry Run (only preview all files that would be uploaded)'
+        required: true
+        default: true
+
+jobs:
+  build_deps:
+    runs-on: ubuntu-20.04
+    if: contains('["maxhbr","nstng", "Neudrino", "tmdzk", "alexzurbonsen", "MoritzThomasHuebner"]', github.actor)
+    steps:
+      - uses: actions/checkout@7884fcad6b5d53d10323aee724dc68d8b9096a2e # pin@v2
+        with:
+          fetch-depth: 0
+      - name: Install fpm
+        shell: bash
+        run: |
+          sudo apt update
+          sudo apt install -y ruby ruby-dev rubygems build-essential
+          sudo gem install --no-document fpm
+      - name: Build nettle
+        shell: bash
+        run: |
+          mkdir ~/magma-packages
+          cd third_party/build && ./build.py nettle --no-install && cp *.deb ~/magma-packages
+      - name: Testing if .deb file can be installed
+        shell: bash
+        run: |
+          sudo apt install -y ~/magma-packages/*.deb
+      - name: Testing is libnettle.so exists.
+        shell: bash
+        run: |
+          FILE=/usr/lib/libnettle.so
+          if test -f "$FILE"; then
+              echo "$FILE exists."
+          else
+              echo "$FILE does not exist. Build failed. Exiting..."
+              exit 2
+          fi
+      - name: Upload zipped nettle.deb to Github
+        uses: actions/upload-artifact@v3
+        with:
+          name: nettle
+          path: ~/magma-packages/
+      - name: Setup JFROG CLI
+        uses: jfrog/setup-jfrog-cli@v2
+        env:
+          JF_ENV_1: ${{ secrets.JFROG_TOKEN }}
+      - name: Dry run - publish debian packages
+        if: ${{ github.event.inputs.dry_run == 'true'}}
+        run: |
+          jf rt u --recursive=false --url=https://artifactory.magmacore.org:443/artifactory/ --detailed-summary \
+            --dry-run \
+            --target-props="deb.distribution=${{ github.event.inputs.distribution }};deb.component=main;deb.architecture=amd64" \
+            "~/magma-packages/(*).deb" ${{ github.event.inputs.repo }}/pool/${{ github.event.inputs.distribution }}/{1}.deb
+      - name: Publish debian packages
+        if: ${{ github.event.inputs.dry_run == 'false' }}
+        run: |
+          jf rt u --recursive=false --url=https://artifactory.magmacore.org:443/artifactory/ --detailed-summary \
+            --target-props="deb.distribution=${{ github.event.inputs.distribution }};deb.component=main;deb.architecture=amd64" \
+            "~/magma-packages/(*).deb" ${{ github.event.inputs.repo }}/pool/${{ github.event.inputs.distribution }}/{1}.deb
```

Comment:
One should probably avoid this superfluous duplication if the only difference is the `--dry-run` option.

A solution could involve [environment variables](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable).
Something similar to the following:
```yaml
steps:
  - name: Set the value
    id: is_dry
    if: ${{ github.event.inputs.dry_run == 'true' }}
    run: |
      echo "IS_DRY='--dry-run'" >> $GITHUB_ENV
  - name: Use the value
    id: publish
    run: |
      jf rt u --all_other_options_here \ 
        "${{ env.IS_DRY }}"
```

Reply:
I made the change as you suggested but it took me while to figure out what was going wrong. I had to remove some of the quotation marks to make it work.